### ISSUE TITLE
NixOS 24.11 Plasma6 ISO is broken

### ISSUE THREAD

**Issue by InSelfControll** [NixOS/nixpkgs#363797]
## Describe the bug

ISO not loading on VM and barely loading on bare metal machine.
The installation failed 7 times yesterday and at 8th time it succeed but half of the system is broken and things like mounts barely work too, crashing the system to emergency mode.

## Steps To Reproduce

1. Download the latest ISO
2. trying to install it - it'll stuck on black screen for 5 minutes
3. Installation will crash

## Expected behavior

Clean install stable OS like the old one was version 23.11/23.05

## Screenshots
Example on VM.
![image](https://github.com/user-attachments/assets/adbe9f25-3f9e-4808-a109-1ef5899c7c53)
stuck like this for more than 5 minutes.
## Additional context
OS unstable currently I somehow succeed to install it on my main system but I'll have to format again since nothing works.
The name of the ISO says plasma6 while the installation itself installs plasma5.

## Notify maintainers

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by RaySlash** [comment]
@InSelfControll  Please be more specific while you write bug report. Link the ISO file you got and write steps to reproduce properly. Without any logs, we are not able to help you or reproduce the same.

Regardless, you should be fine working with any ISO file. You can alternatively make yourself an ISO file if that fits your need as some hardware like WiFi cards might need some extra kernel module such as `wl`. But it should not cause whatever is happening in your case. Please make sure the file you downloaded SHA matches as well.

**Comment by InSelfControll** [comment]
I cannot share logs from the installation.
The steps I share are the steps I did and encountered the issue.

I had to downgrade back to 24.05 because almost all the apps I use barely worked even Nvidia drivers and basic mounts of brtfs / ext4 drives.

I tried the last two isos of version 24.11 and both of them stuck on black screen like in the pic I shared.

On my main computer I successfully loaded the iso after 7 tries that it was stuck on black screen and after that I successfully got to the install screen and it also failed 7 times on the 8th time after I disabled ipv6 and replaced my DNS servers and reformatted the drive 7 times it decided to work but still ended with broken libs that were actually missing when the system booted so I couldn't stay on it and decided to downgrade.

It was a clean install all the times.
On vm the latest iso was stuck on black screen for over 30 minutes, when I tried with version 24.05 it booted in light speed to the installation and then into the OS.

**Comment by bjornfor** [comment]
I verified that the default graphical ISO (which is based on GNOME) works (in virtual-manager), but I see now that you link to the plasma6 version (which I haven't tested).

**Comment by Ben9986** [comment]
Just tested the plasma 24.11 iso you linked in virt-manager QEMU. Did not boot first time for me. 
Got the boot menu, initial stage 1 outputs then stuck on black screen indefinitely.

Stopped the VM then started it again with no changes made and it booted completely fine. And every time after that too.
Note i did not give the VM any persistent storage, just the read only ISO. 

This was the last output of the failed boot, however the exact same messages also appear on the successful boot so likely not related. 
![image](https://github.com/user-attachments/assets/f972b321-0476-464f-a49a-4597d02f0865)

**Comment by InSelfControll** [comment]
Yea but it doesn't suppose to be like that...
I installed it on my PC and it was stuck like it for 10 minutes each time I tried.
I tried 7 times and at the 8th it booted and then the installation itself failed 7 times and succeed only at the 8th time with a crash at the end so clearly it doesn't install all the necessary OS components correctly.

**Comment by bjornfor** [comment]
> https://releases.nixos.org/nixos/24.11/nixos-24.11.710486.4dc2fc4e62db/nixos-plasma6-24.11.710486.4dc2fc4e62db-x86_64-linux.iso

I also get black screen with that plasma6 ISO in virt-manager (with default settings) after that [stage 1 output](https://github.com/NixOS/nixpkgs/issues/363797#issuecomment-2537449061) posted above. Booted the VM twice.

**Comment by JohnRTitor** [comment]
CC @K900, plasma maintainer

**Comment by K900** [comment]
I can't reproduce here.

**Comment by SuperSandro2000** [comment]
Do you have configured some video memory for the VM?

**Comment by InSelfControll** [comment]
> Do you have configured some video memory for the VM?

I'm talking about my main PC not the VM.
On the VM when I tested it I haven't configured anything special, base OS.

**Comment by K900** [comment]
Can you get any sort of log output from your PC?

**Comment by InSelfControll** [comment]
> Can you get any sort of log output from your PC?

Already downgraded to version 24.05 can't send a log.

**Comment by CallMeLaNN** [comment]
This is my first time to try NixOS Plasma 6 from the [Download (Plasma Desktop, 64-bit Intel/AMD)](https://channels.nixos.org/nixos-24.11/latest-nixos-plasma6-x86_64-linux.iso) button 3 days ago, I'm looking around to fix my graphic driver, I'm not sure until I found this.

Downloaded this name: nixos-plasma6-24.11.711349.394571358ce8-x86_64-linux.iso

Flash in USB thumbdrive, boot it on an old PC, select "Installer", receive [error message like above](https://github.com/NixOS/nixpkgs/issues/363797#issuecomment-2537449061) and then just blank screen with "_" for long time.

CTRL+ALT+F1 to view logs. This are some errors I saw
- [Firmware Bug](https://askubuntu.com/questions/984970/firmware-bug-tsc-deadline-disabled-due-to-errata-what-should-i-do-about-thi) which might not related
- nouveau error with some memory addresses
- spice vdagentd with no data available

**Comment by K900** [comment]
Is this Nvidia? Which GPU model?

**Comment by InSelfControll** [comment]
> Is this Nvidia? Which GPU model? 

I'm using an Nvidia GPU.

**Comment by K900** [comment]
Which exact model?

**Comment by InSelfControll** [comment]
> Which exact model? 

Rtx 2070 super but it was also in VM without GPU.

**Comment by K900** [comment]
Those are likely separate issues. Does booting in nomodeset mode work?

**Comment by InSelfControll** [comment]
> Those are likely separate issues. Does booting in nomodeset mode work? 

No, the Nvidia driver was also broken on 24.11 it couldn't detect the GPU correctly so always back to noveau.

I had to downgrade the OS to make it work, only after downgrading and disabling sddm my PC became more stable.

**Comment by CallMeLaNN** [comment]
Mine one nvidia gtx 1060. 

Booting from nomodeset also the same, it's blank if I switch to ctrl+alt+f2. It always hang for a moment when switching to there. 
Using nomodeset, if I restart the desktop-manager svc, I can see login screen (with the word wayland at the bottom) but if I enter, it become black again with "x" at the center. 

Reproduce for a couple of times, it is consistent.

**Comment by jogrie** [comment]
Hi together,
i have the same problem and i got it fixed.

Yesterday i tried to install NixOS with Plasma on a HP ProDesk 400 with an AMD Ryzen 3400G, but the ISO bootet only to the Blank screen with the blinking underscore.

So i decided to boot the Gnome Image which worked fine.

After the installation of NixOS with Plasma 6 i had the same again.
When Plasma should start i got redirectet to the shell Login screen (STRG + ALT + F1)
Switching to the graphical session (STRG + ALT + F2) showed only the blinking underscore.

after a few tries i found the solution.

i made the following change in the nixos configuration
old
services.displayManager.sddm.enable = true;
new
services.displayManager.sddm.wayland.enable = true;

tldr:
setting up sddm for wayland fixed the problem.
services.displayManager.sddm.wayland.enable = true;

**Comment by muggenhor** [comment]
> tldr: setting up sddm for wayland fixed the problem.
```nix
services.displayManager.sddm.wayland.enable = true;
```

Thanks for discovering and reporting this! It fixes, or at least works around, #360678 for me.

For reference: I'm running with an nvidia RTX 2060 & Intel i9-9900K and this config snippet:
```nix
  services.xserver = {
    enable = true;
    videoDrivers = [ "nvidia" ];

    xkb = {
      layout = "us";
      model = "pc104";
      options = "eurosign:5,lv3:ralt_switch";
    };
  };

  services.desktopManager.plasma6.enable = true;
  services.displayManager.sddm.enable = true;
  services.displayManager.sddm.wayland.enable = true; # https://github.com/NixOS/nixpkgs/issues/363797#issuecomment-2558384445

  hardware.graphics.enable = true;

  hardware.nvidia = {
    open = true;
    modesetting.enable = true;
    nvidiaSettings = true;
    package = config.boot.kernelPackages.nvidiaPackages.stable;
  };
```

**Comment by K900** [comment]
SDDM on Wayland is still experimental, but we may want to enable it by default if it works better for people...

**Comment by AggraHub** [comment]
I want to confirm that I had the same problem (amd gpu 6700XT) and booting from the GNOME iso and then changing to sddm in the config did fix the problem and allowed me to use KDE.

**Comment by InSelfControll** [comment]
I really don't understand why the ticket closed.
The issue still accord booting into GNOME is not a solution...

**Comment by K900** [comment]
Are you able to reproduce the issue with the latest 24.11 ISO? On what hardware?

**Comment by farahfinn** [comment]
Looks like the issue is not fixed yet. I'm running the latest unstable version of nixos and kde plasma 6
